src/analytics/Averages.js

```diff
@@ -0,0 +1,199 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { Col, Row, Table, InputGroup, InputGroupAddon, InputGroupText, CustomInput, NavLink } from 'reactstrap';
+
+import { getCmc } from 'utils/Card';
+import { sortIntoGroups, getSorts } from 'utils/Sort';
+import ErrorBoundary from 'components/ErrorBoundary';
+
+const useSortableData = (items, config = null) => {
+  const [sortConfig, setSortConfig] = useState(config);
+
+  const sortedItems = React.useMemo(() => {
+    const sortableItems = [...items];
+    if (sortConfig !== null) {
+      sortableItems.sort((a, b) => {
+        if (a[sortConfig.key] - b[sortConfig.key] < 0) {
+          return sortConfig.direction === 'ascending' ? -1 : 1;
+        }
+        if (a[sortConfig.key] - b[sortConfig.key] > 0) {
+          return sortConfig.direction === 'ascending' ? 1 : -1;
+        }
+        return 0;
+      });
+    }
+    return sortableItems;
+  }, [items, sortConfig]);
+
+  const requestSort = (key) => {
+    let direction = 'descending';
+    if (sortConfig && sortConfig.key === key && sortConfig.direction === 'descending') {
+      direction = 'ascending';
+    }
+    setSortConfig({ key, direction });
+  };
+
+  return { items: sortedItems, requestSort, sortConfig };
+};
+
+const HeaderCell = ({ label, fieldName, sortConfig, requestSort }) => {
+  let icon = '/content/nosort.png';
+
+  if (sortConfig && sortConfig.key === fieldName) {
+    if (sortConfig.direction === 'descending') {
+      icon = '/content/ascending.png';
+    } else {
+      icon = '/content/descending.png';
+    }
+  }
+
+  return (
+    <th scope="col">
+      <NavLink href="#" onClick={() => requestSort(fieldName)}>
+        {label} <img src={icon} className="sortIcon" alt="" />
+      </NavLink>
+    </th>
+  );
+};
+
+HeaderCell.propTypes = {
+  label: PropTypes.string.isRequired,
+  fieldName: PropTypes.string.isRequired,
+  sortConfig: PropTypes.shape({
+    key: PropTypes.string.isRequired,
+    direction: PropTypes.string.isRequired,
+  }).isRequired,
+  requestSort: PropTypes.string.isRequired,
+};
+
+const Averages = ({ cards }) => {
+  const sorts = getSorts();
+  const characteristics = {
+    CMC: getCmc,
+    Power: (card) => parseInt(card.details.power, 10),
+    Toughness: (card) => parseInt(card.details.toughness, 10),
+    Elo: (card) => parseFloat(card.details.elo, 10),
+    Price: (card) => parseFloat(card.details.price),
+    'Price Foil': (card) => parseFloat(card.details.price_foil),
+  };
+
+  const [sort, setSort] = useState('Color');
+  const [Characteristic, setCharacteristic] = useState('CMC');
+
+  const groups = sortIntoGroups(cards, sort);
+
+  const average = (arr) => {
```

Comment:
I'd like these methods to be in a helper file since they can come up more places like the draft bot code. Also needs a version that works with weighted arrays becuase of asfan. I can give you the code for those if you'd like.